### ISSUE TITLE
Updatable description field

### ISSUE THREAD

**Issue by mdering** [tqdm/tqdm#173]
Would it be possible to have an updatable field in a progress bar? For example, I'm training a machine learning model and on each iteration it returns a loss. I tried making this the bar description but it seems like that doesn't update on each iteration. Is there some way to achieve this?

**Comment by lrq3000** [comment]
`tqdm.set_description()` does update at each iteration the description, but the bar display is not refreshed at each iteration by default, to economize cpu cycles. To force the refresh at each iteration, just specify the argument `miniters=1`.

**Comment by mdering** [comment]
How often does it refresh then? Because I see it refresh just once, when I start a new progress bar.

**Comment by lrq3000** [comment]
If `miniters=None` then it will be automatically adjusted to refresh every `mininterval`, the default being 0.1 second. If your bar finishes under 0.1 second, it should refresh at least once to display the completed bar.

**Comment by mdering** [comment]
Well, it definitely does not behave that way. I should mention i'm using the notebook widget. My code:

`for epoch in tqdm_notebook(range(nb_epoch)):
    for x,y in tqdm_notebook(data_gen(train_data),
                             total=3802,leave=False,
                            desc = "loss: {:.4f}".format(lossflt)):
        loss = model.train_on_batch(x,y)
        lossflt = np.asscalar(loss)
        losses.append(loss)`

**Comment by lrq3000** [comment]
Ah you are using the `tqdm_notebook` submodule. Submodules may behave a bit differently (and this one in particular is experimental).
Can you try to set `mininterval=0, miniters=1` ?

**Comment by mdering** [comment]
Sure, I can try that later or tomorrow (running an 8 hour job at the moment).

**Comment by lrq3000** [comment]
Wait you are not using `set_description()`. What is your problem exactly? That the bar is not displayed at all or that the description isn't updated?

**Comment by mdering** [comment]
I'm using desc = "loss: {:.4f}".format(lossflt). The description is not updated. currently it says loss: 0.0197 for each batch, even though the loss will fluctuate on a per-batch basis.

**Comment by lrq3000** [comment]
Ah that's normal then. The `desc` argument sets the initial description, but it doesn't update it (`.format()` is not a callback anyway). You need to update your description inside your loop, like this:

```
for epoch in tqdm_notebook(range(nb_epoch)):
    t = tqdm_notebook(data_gen(train_data),
                             total=3802, leave=False):
    for x,y in t:
        loss = model.train_on_batch(x,y)
        lossflt = np.asscalar(loss)
        t.set_description("loss: {:.4f}".format(lossflt))
        losses.append(loss)
```

**Comment by mdering** [comment]
man, that makes a lot of sense. so much sense I'm embarrassed. Thanks a lot.

**Comment by lrq3000** [comment]
No problem, feel free to ask if you're not sure, `tqdm` is getting so many functionalities that it is easy to get lost :)

BTW I checked `mininterval` and `miniters` with `tqdm_notebook`, they work flawlessly, so you can use them to adjust the timing you want to refresh the display of your bars.
